File: Csharp/LcdNumbers/ScalingRepeater.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Repeat objects and characters according to scaling.</summary>
    public class ScalingRepeater
    {
        public virtual IList<T> Repeat<T>(T element, Scaling scaling)
        {
            if (element == null)
            {
                throw new ArgumentNullException(nameof(element));
            }
            if (scaling == null)
            {
                throw new ArgumentNullException(nameof(scaling));
            }

            IList<T> elements = new List<T>();
            scaling.Times(() => elements.Add(element));
            return elements;
        }

        public virtual string Repeat(char aChar, Scaling scaling)
        {
            if (scaling == null)
            {
                throw new ArgumentNullException(nameof(scaling));
            }

            StringBuilder acc = new StringBuilder();
            scaling.Times(() => acc.Append(aChar));
            return acc.ToString();
        }
    }
}

File: PHP/src/LcdNumbers/ScalingRepeater.php
<?php
namespace LcdNumbers;

/**
 * Repeat objects and characters according to scaling.
 */
class ScalingRepeater
{

    public function repeat($element, Scaling $scaling)
    {
        $elements = [];
        $scaling->times(function () use (&$elements, $element) {
            array_push($elements, $element);
        });
        return $elements;
    }

    public function repeatCharacter($aChar, Scaling $scaling)
    {
        $result = "";
        $scaling->times(function () use (&$result, $aChar) {
            $result = $result . $aChar;
        });
        return $result;
    }
}
